File: pingdart-java/src/main/java/com/pingdart/sdk/services/DatabaseService.java
package com.pingdart.sdk.services;

import com.google.gson.Gson;
import com.google.gson.JsonObject;
import okhttp3.*;

import java.io.IOException;

public class DatabaseService {
    private final OkHttpClient http;
    private final String databaseId;
    private final Gson gson = new Gson();

    public DatabaseService(OkHttpClient http, String databaseId) {
        this.http = http;
        this.databaseId = databaseId;
    }

    private String postRequest(String endpoint, JsonObject data) throws IOException {
        data.addProperty("databaseid", databaseId);
        RequestBody body = RequestBody.create(gson.toJson(data), MediaType.get("application/json; charset=utf-8"));
        Request request = new Request.Builder().url("/" + endpoint).post(body).build();
        try (Response response = http.newCall(request).execute()) {
            return response.body().string();
        }
    }

    public String read(String schema, String table, JsonObject conditions) throws IOException {
        JsonObject data = new JsonObject();
        data.addProperty("tableSchema", schema);
        data.addProperty("tableName", table);
        data.add("conditions", conditions);
        return postRequest("dynamicRead", data);
    }
    
    // Additional CRUD methods follow similar pattern...
}

File: pingdart-dart/lib/src/services/database_service.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

class DatabaseService {
  final http.Client client;
  final String apiKey;
  final String baseUrl;
  final String databaseId;

  DatabaseService({
    required this.client,
    required this.apiKey,
    required this.baseUrl,
    required this.databaseId,
  });

  Future<Map<String, dynamic>> _postRequest(String endpoint, Map<String, dynamic> data) async {
    if (!data.containsKey('databaseid')) {
      data['databaseid'] = databaseId;
    }

    final response = await client.post(
      Uri.parse('$baseUrl$endpoint'),
      headers: {
        'x-api-key': apiKey,
        'Content-Type': 'application/json',
        'X-SDK-Source': 'PingDart-Dart-SDK',
      },
      body: jsonEncode(data),
    );

    return jsonDecode(response.body);
  }

  Future<Map<String, dynamic>> create(String schema, String table, Map<String, dynamic> data, {Map<String, dynamic>? conditions}) {
    return _postRequest('dynamicCreate', {
      'tableSchema': schema,
      'tableName': table,
      'data': data,
      'conditions': conditions ?? {},
    });
  }

  Future<Map<String, dynamic>> read(String schema, String table, {Map<String, dynamic>? conditions}) {
    return _postRequest('dynamicRead', {
      'tableSchema': schema,
      'tableName': table,
      'conditions': conditions ?? {},
    });
  }
}
